sequence_task_1/student/include/impl.h

```diff
@@ -19,32 +22,99 @@ template<class Container>
 class StaticPlaylist
 {
 public:
+
     /** @todo Member traits */
+    using value_type      = typename Container::value_type;
+    using reference       = typename Container::reference;
+    using const_reference = typename Container::const_reference;
+    using iterator        = typename Container::iterator;
+    using const_iterator  = typename Container::const_iterator;
+    using difference_type = typename Container::difference_type;
+    using size_type       = typename Container::size_type;
 
     /** @todo Iterators */
 
+    const_iterator begin()
+    {
+        return m_tracklist.begin();
+    }
+    const_iterator end()
+    {
+        return m_tracklist.end();
+    }
+    const_iterator cbegin()
+    {
+        return m_tracklist.cbegin();
+    }
+    const_iterator cend()
+    {
+        return m_tracklist.cend();
+    }
+
     /** @todo Constructor from any reversible sequence container */
 
+    template <typename T>
+    StaticPlaylist(const T& container)
+    {
+        m_tracklist.assign(container.begin(), container.end());
+    }
+
     /** @todo Assignment from any reversible sequence container */
+    template <typename T>
+    StaticPlaylist& operator=(T &container)
+    {
+        //assigns new songs to the container from the begging;
+        m_tracklist.assign(container.begin(), container.end());
+        return *this;
+    }
+
+    //StaticPlaylist& operator=(const StaticPlaylist &container);
 
     /** @todo Add track from initializer */
     template<class... Args>
-    const Song& play(Args&&... songData);
+    const Song& play(Args&&... songData)
+    {
+        m_tracklist.insert(m_tracklist.end(), {Song(songData)...});
```

Comment:
You already have all needed data for in-place container element construction. Don't create a temporary object!